src/FieldArray.jl

```diff
@@ -31,3 +31,7 @@ similar_type(::Type{A}, ::Type{T}, S::Size) where {N, T, A<:FieldArray{N, T}} =
 _fieldarray_similar_type(A, T, NewSize::S, OldSize::S) where {S} = A
 _fieldarray_similar_type(A, T, NewSize, OldSize) =
     default_similar_type(T, NewSize, length_val(NewSize))
+
+# Convenience constructors for NamedTuple types 
+Base.convert(::Type{NamedTuple}, array::FieldArray) = Base.NamedTuple(array)
```

Comment:
I don't think this is a good idea since `convert` is called implicitly in some cases. Note also that `convert(::Tuple, ::Array)` isn't defined either, which would be similar to this.

Reply:
OK, then let's only define the constructor here.

Reply:
Done!